src/OpenTelemetry/Trace/Builder/TracerProviderBuilderState.cs

```diff
@@ -109,11 +109,20 @@ public void SetResourceBuilder(ResourceBuilder resourceBuilder)
             this.ResourceBuilder = resourceBuilder;
         }
 
-        public void SetSampler(Sampler sampler)
+        public void ConfigureSampler(Action<SamplerBuilder> configure)
         {
-            Debug.Assert(sampler != null, "sampler was null");
+            Debug.Assert(configure != null, "configure was null");
```

Comment:
Shouldn't it be `Guard.ThrowIfNull(configure);`?
The same comment applies to other places.